lib/msf/core/auxiliary/report_summary.rb

```diff
@@ -0,0 +1,132 @@
+# -*- coding: binary -*-
+
+##
+# This file is part of the Metasploit Framework and may be subject to
+# redistribution and commercial restrictions. Please see the Metasploit
+# Framework web site for more information on licensing and terms of use.
+# https://metasploit.com/framework/
+##
+
+module Msf
+  class Auxiliary
+    ###
+    #
+    # This module provides a means to report module summaries
+    #
+    ###
+    module ReportSummary
+      def initialize(info = {})
+        super(info)
+
+        if framework.features.enabled?(Msf::FeatureManager::SHOW_SUCCESSFUL_LOGINS)
+          register_options(
+            [
+              OptBool.new('ShowSuccessfulLogins', [false, 'Outputs a table of successful logins', true]),
+            ]
+          )
+        end
+      end
+
+      # TODO: Fix class variables
+      #   - Need to sync up with Alan on stuff
+      def run
+        @@report = {}
+        super
+        return unless framework.features.enabled?(Msf::FeatureManager::SHOW_SUCCESSFUL_LOGINS)
+        return unless datastore['ShowSuccessfulLogins'] == true
+
+        print_report_summary
+      end
+
+      def create_credential_login(credential_data)
+        credential = credential_data[:core].to_credential
+        @@report[rhost] = { successful_logins: [] }
+        @@report[rhost][:successful_logins] << credential
+        super
+      end
+
+      # I think this will work for our new modules, but I don't know if it'll work
+      # for other modules that might not use 'session_setup' convention;
+      # i.e. you might need to hijack start_session, or create_session, or some other
+      # lower level building block that you'll have access to
+      def start_session(obj, info, ds_merge, crlf = false, sock = nil, sess = nil)
+        result = super
+        @@report[rhost].merge!({ successful_sessions: [] })
+        @@report[rhost][:successful_sessions] << result
+        result
+      end
+
+      private
+
+      def print_report_summary
+        report = @@report
+
+        conditional_verbose_output(report.keys.count)
+
+        logins = report.flat_map { |_k, v| v[:successful_logins] }.compact
+        sessions = report.flat_map { |_k, v| v[:successful_sessions] }.compact
+
+        # TODO - This was previously within the run command that was initially added as a workaround
+        # return results unless framework.features.enabled?(Msf::FeatureManager::MSSQL_SESSION_TYPE)
+
+        print_status("Bruteforce completed, #{logins.size} #{logins.size == 1 ? 'credential was' : 'credentials were'} successful.")
+        if datastore['CreateSession']
+          print_status("#{sessions.size} MySQL #{sessions.size == 1 ? 'session was' : 'sessions were'} opened successfully.")
+        else
+          print_status('You can open an MySQL session with these credentials and %grnCreateSession%clr set to true')
```

Comment:
This message doesn't seem cromulent for other scanners such as ssh

```
msf6 auxiliary(scanner/ssh/ssh_login) > run rhost=127.0.0.1 <------

[*] 127.0.0.1:22 - Starting bruteforce
[*] Error: 127.0.0.1: Metasploit::Framework::LoginScanner::Invalid Cred details can't be blank, Cred details can't be blank (Metasploit::Framework::LoginScanner::SSH)
[*] Scanned 1 of 1 hosts (100% complete) 
[*] Bruteforce completed, 0 credentials were successful. 
[*] 0 MySQL sessions were opened successfully.           <---------
```

Reply:
Whoops, yeah I can make that more generic.